### PR TITLE
adds a cache for public/static

### PR DESCRIPTION
**Pull request by joelhooks**
This is a hacked up version of `plugin-netlify-cache` that only caches public/static and has a two week clearing piece for the folder.

### CODE REVIEW

File: plugins/netlify-cache/gatsby-node.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    )
+    await ensureDir(cachePath)
+    const cacheFiles = await readdir(cachePath)
+    let shouldClear = false
+    const fortnightAway = new Date(Date.now() - 12096e5).getTime()
+    cacheFiles.forEach(file => {
+      try {
+        const stats = fs.statSync(cachePath + '/' + file)
+        const birthTime = new Date(stats.birthtime).getTime()
+        shouldClear = isBefore(birthTime, fortnightAway)
```
Comment (joelhooks) [inline plugins/netlify-cache/gatsby-node.js:105]: if anything is older than a fortnight, the whole folder is cleared 🤤

**Comment by joelhooks** [comment]
Not terribly proud of this, but it should do the job 😂

**Comment by joelhooks** [comment]
Cuts build time in half!

**Review by kentcdodds — APPROVED** [review]
Dude, that's sweeeet 💯

**Pull request merged**